### PR TITLE
secp256k1 ec verify ref #165

### LINKED ISSUE #165: Check signature command should verify secp256k1 EC
**Describe the bug**
Check message command is not complete.

**Environment information:**

- OS: all
- Platform: all
- Firmware Version: b6846cb

**Steps to Reproduce**
This is an internal firmware error detected after review

**Actual behavior**
Signature verification procedure in firmware is like this

- recover pubkey from message and sig
- generate inferred address from pubkey
- inferred address should match address

**Expected behavior**
In `skycoin/skycoin` the three steps above are followed by verifying signature using `secp256k1` EC

### PR DESCRIPTION
**Pull request by stdevAlDen**
Fixes #165

Changes:	
 - Add `verify_pub_key`

Does this change need to mentioned in CHANGELOG.md?	
yes

Requires testing	
yes

Depends on:
 - https://github.com/skycoin/hardware-wallet/pull/252
 - https://github.com/skycoin/hardware-wallet/pull/254

### CODE REVIEW

File: tiny-firmware/firmware/fsm_impl.c [hunk 1]
```diff
@@ -225,6 +225,10 @@ ErrCode_t msgSkycoinCheckMessageSignatureImpl(SkycoinCheckMessageSignature* msg,
             strncpy(failureResp->message, _("Address does not match"), sizeof(failureResp->message));
             failureResp->has_message = true;
             ret = ErrInvalidSignature;
+        } else if (!verify_pub_key(pubkey)) {
+            strncpy(failureResp->message, _("Address vrification failed"), sizeof(failureResp->message));
```
Comment (gz-c) [inline tiny-firmware/firmware/fsm_impl.c]: `vrification` typo

File: tiny-firmware/firmware/fsm_impl.c [hunk 2]
```diff
@@ -225,6 +225,10 @@ ErrCode_t msgSkycoinCheckMessageSignatureImpl(SkycoinCheckMessageSignature* msg,
...
+            failureResp->has_message = true;
+            ret = ErrInvalidSignature;
```
Comment (gz-c) [inline tiny-firmware/firmware/fsm_impl.c]: is that the right `ret` value?

File: skycoin-api/skycoin_crypto.c [hunk 3]
```diff
@@ -31,6 +31,19 @@ static void create_node(const char* seed_str, HDNode* node)
     hdnode_fill_public_key(node);
 }
 
+bool verify_pub_key(const uint8_t* pub_key) {
+    (void)pub_key;
+    char seed_str[256] = "dummy seed";
```
Comment (gz-c) [inline skycoin-api/skycoin_crypto.c]: why does this have initializer text? what is `seed_str`?

Comment (olemis) [reply skycoin-api/skycoin_crypto.c]: If I understood correctly , that's the BIP44 HD path to the HD node used to derive the seed. Even if Skycoin does not implement BIP44 (yet ? see skycoin/skycoin#2247 ) it looks like required field for Trezor libraries .

File: tiny-firmware/firmware/fsm_impl.c (hunk 2 above)
Comment (olemis) [reply tiny-firmware/firmware/fsm_impl.c]: No . Maybe `ErrAddressGeneration` ?

File: skycoin-api/skycoin_crypto.c (hunk 3 above)
Comment (stdevAlDen) [reply skycoin-api/skycoin_crypto.c]: > why does this have initializer text? what is `seed_str`?

I'm not sure why this is used, but functions that require a ecdsa_curve like [`ecdsa_read_pubkey`](https://github.com/skycoin/hardware-wallet/blob/384c508d07b8909c58f9afba5b8677fc409e66bb/skycoin-api/tools/ecdsa.h#L89) get the curve in [this way](https://github.com/skycoin/hardware-wallet/blob/master/skycoin-api/skycoin_check_signature.c#L82-L85). I think that this pr is critical enough to add this to the [review issue](https://github.com/skycoin/hardware-wallet/issues/104).

File: tiny-firmware/firmware/fsm_impl.c (hunk 2 above)
Comment (stdevAlDen) [reply tiny-firmware/firmware/fsm_impl.c]: @gz-c & @olemis right, thank you let me fix

File: skycoin-api/skycoin_crypto.c (hunk 3 above)
Comment (gz-c) [reply skycoin-api/skycoin_crypto.c]: Its not required at all and its a roundabout way of obtaining a reference to the secp256k1 curve parameters, see https://github.com/skycoin/hardware-wallet/pull/254 which fixes it

**Pull request merged**